compiler/rustc_const_eval/src/errors.rs

```diff
@@ -749,158 +1080,1059 @@ impl<'tcx> ReportErrorExt for ValidationErrorInfo<'tcx> {
+                    "pointer",
+                    pointer,
+                )
+            }
+            DanglingPtrNoProvenance { ptr_kind: PointerKind::Ref, pointer } => {
+                eagerly_translate_with_args!(
+                    handler,
+                    const_eval_validation_dangling_ref_no_provenance,
+                    "front_matter",
+                    front_matter,
+                    "pointer",
+                    pointer,
+                )
+            }
+            DanglingPtrOutOfBounds { ptr_kind: PointerKind::Box } => {
+                eagerly_translate_with_args!(
+                    handler,
+                    const_eval_validation_dangling_box_out_of_bounds,
+                    "front_matter",
+                    front_matter,
+                )
+            }
+            DanglingPtrOutOfBounds { ptr_kind: PointerKind::Ref } => {
+                eagerly_translate_with_args!(
+                    handler,
+                    const_eval_validation_dangling_ref_out_of_bounds,
+                    "front_matter",
+                    front_matter,
+                )
+            }
+            DanglingPtrUseAfterFree { ptr_kind: PointerKind::Box } => {
+                eagerly_translate_with_args!(
+                    handler,
+                    const_eval_validation_dangling_box_use_after_free,
+                    "front_matter",
+                    front_matter,
+                )
+            }
+            DanglingPtrUseAfterFree { ptr_kind: PointerKind::Ref } => {
+                eagerly_translate_with_args!(
+                    handler,
+                    const_eval_validation_dangling_ref_use_after_free,
+                    "front_matter",
+                    front_matter,
+                )
+            }
+            InvalidBool { value } => {
+                eagerly_translate_with_args!(
+                    handler,
+                    const_eval_validation_invalid_bool,
+                    "front_matter",
+                    front_matter,
+                    "value",
+                    value,
+                )
+            }
+            InvalidChar { value } => {
+                eagerly_translate_with_args!(
+                    handler,
+                    const_eval_validation_invalid_char,
+                    "front_matter",
+                    front_matter,
+                    "value",
+                    value,
+                )
+            }
+            InvalidFnPtr { value } => {
+                eagerly_translate_with_args!(
+                    handler,
+                    const_eval_validation_invalid_fn_ptr,
+                    "front_matter",
+                    front_matter,
+                    "value",
+                    value,
+                )
+            }
+        }
+    }
+}
+
+impl AddToDiagnostic for ValidationErrorInfoExt2<'_> {
+    fn add_to_diagnostic_with<F>(self, diag: &mut Diagnostic, _: F)
+    where
```

Comment:
`ValidationErrorInfoExt2` is an unhelpful name. I think this should be an associated function for `ValidationErrorInfoExt` named `as_subdiagnostic` and this type can be called `ValidationErrorInfoAsSubdiagnosticExt`. Because the type name would be so long, I think you should seal it in a private module so that no one except `rustc_const_eval::errors` can refer to it.

Reply:
Would you like all `*Ext2` variants to be renamed in a similar fashion?

Reply:
Yes

Reply:
I've gotten rid of the `*Ext` and `*Ext2` confusion and merged it into `*Ext`. Would you still like proceed with the renaming of the types?